src/psyclone/tests/gocean1p0_build.py

```diff
@@ -0,0 +1,139 @@
+
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2017-2018, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors: J. Henrichs, Bureau of Meteorology
+
+
+''' Module containing configuration required to build code generated
+for the Dynamo0p3 API '''
+
+from __future__ import absolute_import, print_function
+import os
+from psyclone_test_utils import Compile, CompileError
+
+
+class GOcean1p0Build(Compile):
+    '''Build class for compiling test files for the GOcean1.0 api. It
+    uses dl_esm_inf as included in the PSyclone distribution.
+    '''
+
+    _infrastructure_built = False
+    _infrastructure_path = ""
+
+    def __init__(self, tmpdir):
+        super(GOcean1p0Build, self).__init__(tmpdir)
+
+        base_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                 "test_files", "gocean1p0")
+        self.base_path = base_path
+
+        # On first instantiation (triggered by conftest.infra_compile)
+        # compile the infrastructure library files.
+        if Compile.TEST_COMPILE and not GOcean1p0Build._infrastructure_built:
+            self._build_infrastructure()
+            GOcean1p0Build._infrastructure_path = str(tmpdir)
```

Comment:
It would make sense for `_build_infrastructure` to be responsible for storing the infrastructure path.